tests/jni_api.rs

```diff
@@ -101,6 +101,19 @@ pub fn is_instance_of_null() {
     ));
 }
 
+#[test]
+pub fn test_same_objects() {
+    const SOURCE_STR: &str = "THIS IS STRING OBJECT";
+    let env = attach_current_thread();
+    let string = env.new_string(SOURCE_STR).unwrap();
+    let ref_from_string = unwrap(&env, env.new_local_ref::<JObject>(string.into()));
+    assert!(unwrap(&env, env.is_same_object(string, ref_from_string)));
+    unwrap(&env, env.delete_local_ref(ref_from_string));
+
+    let same_src_str = env.new_string(SOURCE_STR).unwrap();
```

Comment:
These are different test-cases, I think it is appropriate to put them into separate test methods, e.g.:
- test_same_object_diff_references, 
- test_same_object_same_reference
- test_not_same_object
- test_not_same_object_null